### DISCUSSION TITLE
Cache rejected query

### DISCUSSION THREAD

**Discussion by ThomasRutzer** [reduxjs/redux-toolkit#3205] [Q&A]
If I may have a previously rejected query ( stored as `status: rejected` because of http error 400) in the redux-store, is there any chance to still use cache behaviour for that if an upcoming query would match?

**Comment by markerikson** [comment]
Not sure I understand the question. Can you give more details of what you're trying to do?

**Reply by ThomasRutzer** [reply]
Ok, I'll give my best: 

First, I am working with an API, I don't have much control over.
So this API gives me a status 400 for a POST request with body `{value: "12345" }`. This is then stored in my redux-store api section as something like this:
```
{
    "status": "rejected",
    "endpointName": "myEndpoint",
    "requestId": <any-id>,
    "originalArgs": "12345",
    "startedTimeStamp": <any-timestamp>,
    "error": {
        "status": 400,
        "data": {
            ...
        }
    }
}

```

If now, because of the behaviour of this API, another request to this Endpoint with `{value: "12345" }` is triggered in my application, I'd like the cache behaviour of RTK query to step in and preventing a network call. Is this understandable?

**Reply by markerikson** [reply]
Hmm. Are you effectively wanting to treat that 400 status as a "succeeded" result that actually gets cached and used as real data?

**Reply by ThomasRutzer** [reply]
Yes, something like this, if possible…I mean I suppose I could manipulate this cache entry myself? But this sounds like a bad thing to do…

**Reply by markerikson** [reply]
I don't have an exact answer off the top of my head, but: it seems likely that you could do this by writing a custom `baseQuery` that inspects the response status and contents, or some other kind of a response transformation.

**Reply by ThomasRutzer** [reply]
Hm, actually [`validateStatus`](https://redux-toolkit.js.org/rtk-query/api/fetchBaseQuery#handling-non-standard-response-status-codes) helped…
